shimming-toolbox/shimmingtoolbox/masking/mask_utils.py

```diff
@@ -47,29 +59,39 @@ def resample_mask(nii_mask_from, nii_target, from_slices=None, dilation_kernel='
     nii_mask_target = resample_from_to(nii_mask, nii_target, order=1, mode='grid-constant', cval=0)
     # Resample the full mask onto nii_target
     nii_full_mask_target = resample_from_to(nii_mask_from, nii_target, order=0, mode='grid-constant', cval=0)
-    # TODO: Deal with soft mask
-    # Find highest value and stretch to 1
-    # Look into dilation of soft mask
 
     # Dilate the mask to add more pixels in particular directions
-    mask_dilated = modify_binary_mask(nii_mask_target.get_fdata(), dilation_kernel, dilation_size, 'dilate')
+    if np.array_equal(np.unique(nii_mask_from.get_fdata()), [0, 1]) or nii_mask_from.get_fdata().dtype == bool:
+        mask_dilated = modify_binary_mask(nii_mask_target.get_fdata(), dilation_kernel, dilation_size, 'dilate')
+    else :
+        previous_mask = np.array(nii_mask_target.get_fdata(), dtype=float)
+        kernel = ball(1)
+        for _ in range(dilation_size//2):
+            max_filter = maximum_filter(previous_mask, footprint=kernel, mode='grid-constant')
```

Comment:
Maybe this is sufficient (definetly faster)
```suggestion
       # Set the ball radius to dilation_size//2 or another appropriate value
      kernel = ball(dilation_size//2)
      max_filter = maximum_filter(nii_mask_target.get_fdata(), footprint=kernel, mode='grid-constant')
```

Reply:
Even if it is faster, the kernel size needs to be ball(1), because in a softmask where the voxels in the center are closer to 1 and those in the periphery are closer to 0, a large kernel size would pick those higher values and put a value too high for the voxels added in the dilation. What we want is to "copy" the intensity of the last non-zero voxels (the "edge" voxels)

Reply:
A faster method as been implemented in commit [47a8a30](https://github.com/shimming-toolbox/shimming-toolbox/pull/581/commits/47a8a308610f7b769f27bc9066ab3a684d4e7f94).